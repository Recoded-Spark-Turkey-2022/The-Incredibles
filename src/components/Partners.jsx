import React from 'react';
import Partner from '../assets/pics/partner.svg';

function Partners() {
  const partners = [
    'recoded',
    'recoded',
    'recoded',
    'recoded',
    'recoded',
    'recoded',
    'recoded',
  ];
  return (
    <div className="pt-8 max-lg:pt-0">
      <h1 className="text-center font-bold text-3xl">Our Partners</h1>
      <div className="flex justify-center flex-col overflow-hidden px-4">
        <div className="w-full flex justify-center">
          {partners.map((par, index) => (
            <img
              key={index}
              src={Partner}
              alt="partner-logo"
              className="w-24 grow m-10"
            />
          ))}
        </div>
        <div className="w-full hidden max-lg:flex justify-center pb-8">
          {partners.map((par) => (
            <button
              key={par}
              data-bs-target="#carouselDarkVariant"
              data-bs-slide-to="0"
              className="rounded-full inline-block p-1 mx-1 border-solid border-2 border-cyan-600 focus:bg-cyan-600"
              aria-current="true"
              aria-label="Slide 1"
            ></button>
          ))}
        </div>
      </div>
    </div>
  );
}

export default Partners;
