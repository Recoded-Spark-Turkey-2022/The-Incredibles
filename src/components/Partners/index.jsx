import React from 'react';
import partner1 from '../../assets/pics/homepage/partners/n0.png';
import partner2 from '../../assets/pics/homepage/partners/n1.jpeg';
import partner3 from '../../assets/pics/homepage/partners/n2.png';
import partner4 from '../../assets/pics/homepage/partners/n3.png';
import partner5 from '../../assets/pics/homepage/partners/n4.jpeg';
import partner6 from '../../assets/pics/homepage/partners/n5.png';
import partner7 from '../../assets/pics/homepage/partners/n6.jpeg';
import partner8 from '../../assets/pics/homepage/partners/n7.jpeg';
import partner9 from '../../assets/pics/homepage/partners/n8.jpeg';
import Slider from 'react-slick';
import '../../style/slick.css';
import '../../style/slick-theme.css';
import Container from '../Container';
import { useTranslation } from 'react-i18next';

function Partners() {
  const [t] = useTranslation();
  const partners = [
    partner1,
    partner2,
    partner3,
    partner4,
    partner5,
    partner6,
    partner7,
    partner8,
    partner9,
  ];
  const settings = {
    dots: true,
    infinite: true,
    speed: 2000,
    slidesToShow: 6,
    slidesToScroll: 3,
    autoplay: true,
    cssEase: 'linear',
    autoplaySpeed: 5000,
    swipeToSlide: true,
    responsive: [
      {
        breakpoint: 1024,
        settings: {
          slidesToShow: 5,
          slidesToScroll: 1,
          infinite: true,
          dots: false,
          autoplaySpeed: 1000,
        },
      },
      {
        breakpoint: 767,
        settings: {
          slidesToShow: 3,
          slidesToScroll: 1,
          infinite: true,
          dots: false,
          autoplaySpeed: 1000,
        },
      },
      {
        breakpoint: 450,
        settings: {
          slidesToShow: 1,
          slidesToScroll: 1,
          infinite: true,
          dots: false,
          autoplaySpeed: 1000,
        },
      },
    ],
  };
  return (
    <Container>
      <div className=" my-10">
        <h1 className="text-center font-bold text-3xl">{t('home.partner')}</h1>

        <div className="max-w-screen-xl m-auto flex-row p-7">
          <Slider {...settings} className="py-5">
            {partners.map((par, i) => (
              <div key={i} className="lg:px-10 px-5 ">
                <img src={par} alt="partner-logo" />
              </div>
            ))}
          </Slider>
        </div>
      </div>
    </Container>
  );
}

export default Partners;
