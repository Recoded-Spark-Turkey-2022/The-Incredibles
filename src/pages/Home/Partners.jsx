import React from 'react';
import Partner from '../../assets/pics/homepage/partner.svg';
import { useTranslation } from 'react-i18next';

function Partners() {
  const [t, i18n] = useTranslation();
  const partners = [
    'recoded',
    'recoded',
    'recoded',
    'recoded',
    'recoded',
    'recoded',
    'recoded',
  ];
  return (
    <div className="pt-8 max-lg:pt-0">
      <h1 className="text-center font-bold text-3xl">{t('home.partner')}</h1>
      <div className="flex justify-center flex-col overflow-hidden px-4">
        <div className="w-full flex justify-center">
          {partners.map((par, index) => (
            <img
              key={index}
              src={Partner}
              alt="partner-logo"
              className="w-24 grow m-10"
            />
          ))}
        </div>
        <div className="w-full hidden max-lg:flex justify-center pb-8">
          {partners.map((par, i) => (
            <button
              key={i}
              data-bs-target="#carouselDarkVariant"
              data-bs-slide-to="0"
              className="rounded-full inline-block p-1 mx-1 border-solid border-2 border-cyan-600 focus:bg-cyan-600"
              aria-current="true"
              aria-label="Slide 1"
            ></button>
          ))}
        </div>
      </div>
    </div>
  );
}

export default Partners;
