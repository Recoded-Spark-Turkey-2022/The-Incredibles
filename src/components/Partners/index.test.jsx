import renderer from 'react-test-renderer';
import Partners from '.';

describe('Partners', () =>
  it('renders correctly', () => {
    const domTree = renderer.create(<Partners />).toJSON();
    expect(domTree).toMatchSnapshot();
  }));
