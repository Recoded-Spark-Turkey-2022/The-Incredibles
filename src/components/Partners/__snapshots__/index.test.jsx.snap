// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Partners renders correctly 1`] = `
<div
  className="container max-w-screen-xl m-auto p-10 max-md:px-5"
>
  <div
    className=" my-10"
  >
    <h1
      className="text-center font-bold text-3xl"
    >
      home.partner
    </h1>
    <div
      className="max-w-screen-xl m-auto flex-row p-7"
    >
      <div
        className="slick-slider py-5 slick-initialized"
        dir="ltr"
      >
        <button
          className="slick-arrow slick-prev"
          data-role="none"
          onClick={[Function]}
          style={
            Object {
              "display": "block",
            }
          }
          type="button"
        >
           
          Previous
        </button>
        <div
          className="slick-list"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={null}
          onMouseMove={null}
          onMouseUp={[Function]}
          onTouchCancel={null}
          onTouchEnd={[Function]}
          onTouchMove={null}
          onTouchStart={[Function]}
          style={Object {}}
        >
          <div
            className="slick-track"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
            style={
              Object {
                "WebkitTransform": "translate3d(0px, 0px, 0px)",
                "WebkitTransition": "",
                "msTransform": "translateX(0px)",
                "opacity": 1,
                "transform": "translate3d(0px, 0px, 0px)",
                "transition": "",
              }
            }
          >
            <div
              aria-hidden={true}
              className="slick-slide slick-cloned"
              data-index={-6}
              onClick={[Function]}
              style={
                Object {
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <div
                  className="lg:px-10 px-5 "
                  style={
                    Object {
                      "display": "inline-block",
                      "width": "100%",
                    }
                  }
                  tabIndex={-1}
                >
                  <img
                    alt="partner-logo"
                    className="w-[120px] m-auto"
                    src="n3.png"
                  />
                </div>
              </div>
            </div>
            <div
              aria-hidden={true}
              className="slick-slide slick-cloned"
              data-index={-5}
              onClick={[Function]}
              style={
                Object {
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <div
                  className="lg:px-10 px-5 "
                  style={
                    Object {
                      "display": "inline-block",
                      "width": "100%",
                    }
                  }
                  tabIndex={-1}
                >
                  <img
                    alt="partner-logo"
                    className="w-[120px] m-auto"
                    src="n4.jpeg"
                  />
                </div>
              </div>
            </div>
            <div
              aria-hidden={true}
              className="slick-slide slick-cloned"
              data-index={-4}
              onClick={[Function]}
              style={
                Object {
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <div
                  className="lg:px-10 px-5 "
                  style={
                    Object {
                      "display": "inline-block",
                      "width": "100%",
                    }
                  }
                  tabIndex={-1}
                >
                  <img
                    alt="partner-logo"
                    className="w-[120px] m-auto"
                    src="n5.png"
                  />
                </div>
              </div>
            </div>
            <div
              aria-hidden={true}
              className="slick-slide slick-cloned"
              data-index={-3}
              onClick={[Function]}
              style={
                Object {
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <div
                  className="lg:px-10 px-5 "
                  style={
                    Object {
                      "display": "inline-block",
                      "width": "100%",
                    }
                  }
                  tabIndex={-1}
                >
                  <img
                    alt="partner-logo"
                    className="w-[120px] m-auto"
                    src="n6.jpeg"
                  />
                </div>
              </div>
            </div>
            <div
              aria-hidden={true}
              className="slick-slide slick-cloned"
              data-index={-2}
              onClick={[Function]}
              style={
                Object {
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <div
                  className="lg:px-10 px-5 "
                  style={
                    Object {
                      "display": "inline-block",
                      "width": "100%",
                    }
                  }
                  tabIndex={-1}
                >
                  <img
                    alt="partner-logo"
                    className="w-[120px] m-auto"
                    src="n7.jpeg"
                  />
                </div>
              </div>
            </div>
            <div
              aria-hidden={true}
              className="slick-slide slick-cloned"
              data-index={-1}
              onClick={[Function]}
              style={
                Object {
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <div
                  className="lg:px-10 px-5 "
                  style={
                    Object {
                      "display": "inline-block",
                      "width": "100%",
                    }
                  }
                  tabIndex={-1}
                >
                  <img
                    alt="partner-logo"
                    className="w-[120px] m-auto"
                    src="n8.jpeg"
                  />
                </div>
              </div>
            </div>
            <div
              aria-hidden={false}
              className="slick-slide slick-active slick-current"
              data-index={0}
              onClick={[Function]}
              style={
                Object {
                  "outline": "none",
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <div
                  className="lg:px-10 px-5 "
                  style={
                    Object {
                      "display": "inline-block",
                      "width": "100%",
                    }
                  }
                  tabIndex={-1}
                >
                  <img
                    alt="partner-logo"
                    className="w-[120px] m-auto"
                    src="n0.png"
                  />
                </div>
              </div>
            </div>
            <div
              aria-hidden={false}
              className="slick-slide slick-active"
              data-index={1}
              onClick={[Function]}
              style={
                Object {
                  "outline": "none",
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <div
                  className="lg:px-10 px-5 "
                  style={
                    Object {
                      "display": "inline-block",
                      "width": "100%",
                    }
                  }
                  tabIndex={-1}
                >
                  <img
                    alt="partner-logo"
                    className="w-[120px] m-auto"
                    src="n1.jpeg"
                  />
                </div>
              </div>
            </div>
            <div
              aria-hidden={false}
              className="slick-slide slick-active"
              data-index={2}
              onClick={[Function]}
              style={
                Object {
                  "outline": "none",
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <div
                  className="lg:px-10 px-5 "
                  style={
                    Object {
                      "display": "inline-block",
                      "width": "100%",
                    }
                  }
                  tabIndex={-1}
                >
                  <img
                    alt="partner-logo"
                    className="w-[120px] m-auto"
                    src="n2.png"
                  />
                </div>
              </div>
            </div>
            <div
              aria-hidden={false}
              className="slick-slide slick-active"
              data-index={3}
              onClick={[Function]}
              style={
                Object {
                  "outline": "none",
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <div
                  className="lg:px-10 px-5 "
                  style={
                    Object {
                      "display": "inline-block",
                      "width": "100%",
                    }
                  }
                  tabIndex={-1}
                >
                  <img
                    alt="partner-logo"
                    className="w-[120px] m-auto"
                    src="n3.png"
                  />
                </div>
              </div>
            </div>
            <div
              aria-hidden={false}
              className="slick-slide slick-active"
              data-index={4}
              onClick={[Function]}
              style={
                Object {
                  "outline": "none",
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <div
                  className="lg:px-10 px-5 "
                  style={
                    Object {
                      "display": "inline-block",
                      "width": "100%",
                    }
                  }
                  tabIndex={-1}
                >
                  <img
                    alt="partner-logo"
                    className="w-[120px] m-auto"
                    src="n4.jpeg"
                  />
                </div>
              </div>
            </div>
            <div
              aria-hidden={false}
              className="slick-slide slick-active"
              data-index={5}
              onClick={[Function]}
              style={
                Object {
                  "outline": "none",
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <div
                  className="lg:px-10 px-5 "
                  style={
                    Object {
                      "display": "inline-block",
                      "width": "100%",
                    }
                  }
                  tabIndex={-1}
                >
                  <img
                    alt="partner-logo"
                    className="w-[120px] m-auto"
                    src="n5.png"
                  />
                </div>
              </div>
            </div>
            <div
              aria-hidden={true}
              className="slick-slide"
              data-index={6}
              onClick={[Function]}
              style={
                Object {
                  "outline": "none",
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <div
                  className="lg:px-10 px-5 "
                  style={
                    Object {
                      "display": "inline-block",
                      "width": "100%",
                    }
                  }
                  tabIndex={-1}
                >
                  <img
                    alt="partner-logo"
                    className="w-[120px] m-auto"
                    src="n6.jpeg"
                  />
                </div>
              </div>
            </div>
            <div
              aria-hidden={true}
              className="slick-slide"
              data-index={7}
              onClick={[Function]}
              style={
                Object {
                  "outline": "none",
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <div
                  className="lg:px-10 px-5 "
                  style={
                    Object {
                      "display": "inline-block",
                      "width": "100%",
                    }
                  }
                  tabIndex={-1}
                >
                  <img
                    alt="partner-logo"
                    className="w-[120px] m-auto"
                    src="n7.jpeg"
                  />
                </div>
              </div>
            </div>
            <div
              aria-hidden={true}
              className="slick-slide"
              data-index={8}
              onClick={[Function]}
              style={
                Object {
                  "outline": "none",
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <div
                  className="lg:px-10 px-5 "
                  style={
                    Object {
                      "display": "inline-block",
                      "width": "100%",
                    }
                  }
                  tabIndex={-1}
                >
                  <img
                    alt="partner-logo"
                    className="w-[120px] m-auto"
                    src="n8.jpeg"
                  />
                </div>
              </div>
            </div>
            <div
              aria-hidden={true}
              className="slick-slide slick-cloned"
              data-index={9}
              onClick={[Function]}
              style={
                Object {
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <div
                  className="lg:px-10 px-5 "
                  style={
                    Object {
                      "display": "inline-block",
                      "width": "100%",
                    }
                  }
                  tabIndex={-1}
                >
                  <img
                    alt="partner-logo"
                    className="w-[120px] m-auto"
                    src="n0.png"
                  />
                </div>
              </div>
            </div>
            <div
              aria-hidden={true}
              className="slick-slide slick-cloned"
              data-index={10}
              onClick={[Function]}
              style={
                Object {
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <div
                  className="lg:px-10 px-5 "
                  style={
                    Object {
                      "display": "inline-block",
                      "width": "100%",
                    }
                  }
                  tabIndex={-1}
                >
                  <img
                    alt="partner-logo"
                    className="w-[120px] m-auto"
                    src="n1.jpeg"
                  />
                </div>
              </div>
            </div>
            <div
              aria-hidden={true}
              className="slick-slide slick-cloned"
              data-index={11}
              onClick={[Function]}
              style={
                Object {
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <div
                  className="lg:px-10 px-5 "
                  style={
                    Object {
                      "display": "inline-block",
                      "width": "100%",
                    }
                  }
                  tabIndex={-1}
                >
                  <img
                    alt="partner-logo"
                    className="w-[120px] m-auto"
                    src="n2.png"
                  />
                </div>
              </div>
            </div>
            <div
              aria-hidden={true}
              className="slick-slide slick-cloned"
              data-index={12}
              onClick={[Function]}
              style={
                Object {
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <div
                  className="lg:px-10 px-5 "
                  style={
                    Object {
                      "display": "inline-block",
                      "width": "100%",
                    }
                  }
                  tabIndex={-1}
                >
                  <img
                    alt="partner-logo"
                    className="w-[120px] m-auto"
                    src="n3.png"
                  />
                </div>
              </div>
            </div>
            <div
              aria-hidden={true}
              className="slick-slide slick-cloned"
              data-index={13}
              onClick={[Function]}
              style={
                Object {
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <div
                  className="lg:px-10 px-5 "
                  style={
                    Object {
                      "display": "inline-block",
                      "width": "100%",
                    }
                  }
                  tabIndex={-1}
                >
                  <img
                    alt="partner-logo"
                    className="w-[120px] m-auto"
                    src="n4.jpeg"
                  />
                </div>
              </div>
            </div>
            <div
              aria-hidden={true}
              className="slick-slide slick-cloned"
              data-index={14}
              onClick={[Function]}
              style={
                Object {
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <div
                  className="lg:px-10 px-5 "
                  style={
                    Object {
                      "display": "inline-block",
                      "width": "100%",
                    }
                  }
                  tabIndex={-1}
                >
                  <img
                    alt="partner-logo"
                    className="w-[120px] m-auto"
                    src="n5.png"
                  />
                </div>
              </div>
            </div>
            <div
              aria-hidden={true}
              className="slick-slide slick-cloned"
              data-index={15}
              onClick={[Function]}
              style={
                Object {
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <div
                  className="lg:px-10 px-5 "
                  style={
                    Object {
                      "display": "inline-block",
                      "width": "100%",
                    }
                  }
                  tabIndex={-1}
                >
                  <img
                    alt="partner-logo"
                    className="w-[120px] m-auto"
                    src="n6.jpeg"
                  />
                </div>
              </div>
            </div>
            <div
              aria-hidden={true}
              className="slick-slide slick-cloned"
              data-index={16}
              onClick={[Function]}
              style={
                Object {
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <div
                  className="lg:px-10 px-5 "
                  style={
                    Object {
                      "display": "inline-block",
                      "width": "100%",
                    }
                  }
                  tabIndex={-1}
                >
                  <img
                    alt="partner-logo"
                    className="w-[120px] m-auto"
                    src="n7.jpeg"
                  />
                </div>
              </div>
            </div>
            <div
              aria-hidden={true}
              className="slick-slide slick-cloned"
              data-index={17}
              onClick={[Function]}
              style={
                Object {
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <div
                  className="lg:px-10 px-5 "
                  style={
                    Object {
                      "display": "inline-block",
                      "width": "100%",
                    }
                  }
                  tabIndex={-1}
                >
                  <img
                    alt="partner-logo"
                    className="w-[120px] m-auto"
                    src="n8.jpeg"
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <button
          className="slick-arrow slick-next"
          data-role="none"
          onClick={[Function]}
          style={
            Object {
              "display": "block",
            }
          }
          type="button"
        >
           
          Next
        </button>
        <ul
          className="slick-dots"
          onMouseEnter={null}
          onMouseLeave={null}
          onMouseOver={null}
          style={
            Object {
              "display": "block",
            }
          }
        >
          <li
            className="slick-active"
          >
            <button
              onClick={[Function]}
            >
              1
            </button>
          </li>
          <li
            className=""
          >
            <button
              onClick={[Function]}
            >
              2
            </button>
          </li>
          <li
            className=""
          >
            <button
              onClick={[Function]}
            >
              3
            </button>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
